### PR TITLE
feat: bail in generate when Data Proxy is used with Interactive Transactions

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by Jolg42 — COMMENTED** [review]
I don't think it's always valid to assume that someone is using the Data Proxy (with Interactive transactions) if the following previewFeatures flags are found `'dataProxy', 'interactiveTransactions'`

A user could have these flags and actually not use the Data Proxy nor Interactive Transactions

**Comment by cebuladev** [comment]
If they don't use them, then removing the flag shouldn't be an issue. I know this may be slightly breaking but breaking on the build step, not in runtime, and given the current usage we think that it's better to bail and communicate there rather than in the runtime.

File: packages/sdk/src/get-generators/utils/check-feature-flags/forbiddenTransactionsWithProxyFlagMessage.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import chalk from 'chalk'
+import { highlightDatamodel } from '../../../highlight/highlight'
+import { link } from '../../../link'
+
+export const forbiddenTransactionsWithProxyFlagMessage = `\nInteractive Transactions are not supported in the Data Proxy (${chalk.green(
+  'dataProxy',
+)} feature flag).
+Skip the ${chalk.red('interactiveTransactions')} flag:
```
Comment (Jolg42) [inline packages/sdk/src/get-generators/utils/check-feature-flags/forbiddenTransactionsWithProxyFlagMessage.ts:8]:
```suggestion
Remove ${chalk.red('interactiveTransactions')} from previewFeatures, example:
```

File: packages/sdk/src/get-generators/utils/check-feature-flags/forbiddenTransactionsWithProxyFlagMessage.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+${chalk.bold(
+  highlightDatamodel(`generator client {
+    provider = "prisma-client-js"
+    previewFeatures = ["dataProxy"]
+}`),
+)}
+`
+
+// TODO
```
Comment (Jolg42) [inline packages/sdk/src/get-generators/utils/check-feature-flags/forbiddenTransactionsWithProxyFlagMessage.ts:18]: Todo?

Comment (cebuladev) [reply packages/sdk/src/get-generators/utils/check-feature-flags/forbiddenTransactionsWithProxyFlagMessage.ts:18]: It's a draft PR - we're still agreeing on the communication, there will be a link. Update coming soon.

**Comment by cebuladev** [comment]
@Jolg42 updated to incorporate your suggestions and Alberto's input. PTAL :pray:

**Comment by Jolg42** [comment]
Nice one!

Could you add a small test, example https://github.com/prisma/prisma/blob/main/packages/sdk/src/__tests__/getGenerators/getGenerators.test.ts#L615

**Review by millsp — APPROVED** [review]

**Review by Jolg42 — APPROVED** [review]
Thanks, nice one!

**Pull request merged**